tools/rpkg/src/scan.cpp

```diff
@@ -166,10 +166,19 @@ static unique_ptr<FunctionData> DataFrameScanBind(ClientContext &context, TableF
 			coldata_ptr = (data_ptr_t)INTEGER_POINTER(coldata);
 			break;
 		case RType::DATE:
-		case RType::DATE_INTEGER:
+			if (!IS_NUMERIC(coldata)) {
+				cpp11::stop("DATE_INTEGER should really be integer");
```

Comment:
"DATE should really be integer" would help narrow down failures.

Reply:
There is an oversight here, the check should be IS_INTEGER